src/main/java/service/impl/DataConverterImpl.java

```
@@ -0,0 +1,20 @@
+package service.impl;
+
+import java.util.List;
+import model.FruitTransaction;
+import service.DataConverter;
+import service.Parser;
+
+public class DataConverterImpl implements DataConverter {
+    private Parser<FruitTransaction> parser = new ParserImpl();
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputData) {
+        if (!inputData.get(0).matches("^[bprs],[a-zA-Z]+,\\d+$")) {
```

Checklist violation: The method should always skip the first line (header), regardless of its content, and should throw a RuntimeException if the input list is null or empty. Also, malformed/invalid lines should not be silently skipped; they should cause a RuntimeException with an informative message. (See checklist: input validation, header skipping, and error handling.)